src/pip/_internal/cli/req_command.py

```diff
@@ -158,6 +160,30 @@ def handle_pip_version_check(self, options):
 ]
 
 
+def warn_if_run_as_root():
+    # type: () -> None
+    """Output a warning for sudo users on Unix.
+
+    In a virtual environment, sudo pip still writes to virtualenv.
+    On Windows, users may run pip as Administrator without issues.
+    This warning only applies to Unix root users outside of virtualenv.
+    """
+    if running_under_virtualenv():
+        return
+    if not hasattr(os, "getuid"):
+        return
+    if sys.platform == "win32" or sys.platform == "cygwin":
```

Comment:
An inline comment about why we're not using `WINDOWS` here would be helpful:

https://github.com/pypa/pip/blob/a48ad5385b234097d51283b08c3d933fd81ef534/src/pip/_internal/utils/compat.py#L164

Reply:
```
         return
+    # On Windows, there are no "system managed" Python packages. Installing as
+    # Administrator via pip is the correct way of updating system environments.
     if sys.platform == "win32" or sys.platform == "cygwin":
         return
```

Reply:
I think we should also mention that we're not using utils.compat.WINDOWS because of mypy here.

Reply:
```
    if not hasattr(os, "getuid"):                                                                   
        return                                                                                      
    # On Windows, there are no "system managed" Python packages. Installing as                      
    # Administrator via pip is the correct way of updating system environments.                     
    #                                                                                               
    # We choose sys.platform over utils.compat.WINDOWS here to enable Mypy platform                 
    # checks: https://mypy.readthedocs.io/en/stable/common_issues.html                              
    if sys.platform == "win32" or sys.platform == "cygwin":                                         
        return                                                                                      
    if sys.platform == "darwin" or sys.platform == "linux":                                         
        if os.getuid() != 0:                                                                        
            return 
```